lib/mix/lib/mix/dep/lock.ex

```diff
@@ -32,48 +32,74 @@ defmodule Mix.Dep.Lock do
   """
   @spec read() :: map
   def read() do
-    case File.read(lockfile()) do
-      {:ok, info} ->
-        assert_no_merge_conflicts_in_lockfile(lockfile(), info)
+    lockfile = lockfile()
 
-        case Code.eval_string(info, [], file: lockfile()) do
-          {lock, _binding} when is_map(lock) -> lock
-          {_, _binding} -> %{}
-        end
+    case File.read(lockfile) do
+      {:ok, content} ->
+        read_lock(content, lockfile)
 
       {:error, _} ->
         %{}
     end
   end
 
+  @conflict_regex ~r/<{7} .+\n|\|{7} .+\n[\w\W\s]+={7}\n|={7}\n|>{7} .+\n/
+
+  defp read_lock(content, lockfile) do
+    if String.contains?(content, ~w(<<<<<<< ======= >>>>>>>)) do
```

Comment:
In this particular case, I'd much rather go with `["<<<<<<<", "=======", ">>>>>>>"]` instead of the `~w()` sigil because I find the combination of `()`, `<>` and "hidden" space a bit confusing.